src/common/page-click.service.ts

```diff
@@ -0,0 +1,19 @@
+import {Injectable, EventEmitter} from 'angular2/core';
+
+@Injectable()
+export class NglPageClickService {
+  private clickEmitter: EventEmitter<any> = new EventEmitter<any>();
+
+  constructor() {
+    document.querySelector('html').addEventListener('click', (event: any) => this.clickEmitter.next(event), true);
+  }
+
+  subscribe(callback: (_: any) => void, filter: (_: any) => boolean) {
```

Comment:
I understand that you try to have a single global event listener for ngl components to subscribe and this is an interesting approach.

A more preferable way of approaching this issue would be to utilize  angular's event listeners by adding `'(document: click)': 'handlePageClickEvent( $event )',` in the `host` attribute of the `Directive` decorator.

There are no redundant event listeners as `angular2/system` and `angula2/zone` modules take care of this by using a *synthetic* event system in order to maintain performance levels.